### PR TITLE
[Merged by Bors] - Reduce some EE and builder related ERRO logs to WARN

### PR DESCRIPTION
**Pull request by paulhauner**
## Issue Addressed

NA

## Proposed Changes

Our `ERRO` stream has been rather noisy since the merge due to some unexpected behaviours of builders and EEs. Now that we've been running post-merge for a while, I think we can drop some of these `ERRO` to `WARN` so we're not "crying wolf".

The modified logs are:

#### `ERRO Execution engine call failed`

I'm seeing this quite frequently on Geth nodes. They seem to timeout when they're busy and it rarely indicates a serious issue. We also have logging across block import, fork choice updating and payload production that raise `ERRO` or `CRIT` when the EE times out, so I think we're not at risk of silencing actual issues.

#### `ERRO "Builder failed to reveal payload"`

In #3775 we reduced this log from `CRIT` to `ERRO` since it's common for builders to fail to reveal the block to the producer directly whilst still broadcasting it to the networ. I think it's worth dropping this to `WARN` since it's rarely interesting.

I elected to stay with `WARN` since I really do wish builders would fulfill their API promises by returning the block to us. Perhaps I'm just being pedantic here, I could be convinced otherwise.

#### `ERRO "Relay error when registering validator(s)"`

It seems like builders and/or mev-boost struggle to handle heavy loads of validator registrations. I haven't observed issues with validators not actually being registered, but I see timeouts on these endpoints many times a day. It doesn't seem like this `ERRO` is worth it.

#### `ERRO Error fetching block for peer     ExecutionLayerErrorPayloadReconstruction`

This means we failed to respond to a peer on the P2P network with a block they requested because of an error in the `execution_layer`. It's very common to see timeouts or incomplete responses on this endpoint whilst the EE is busy and I don't think it's important enough for an `ERRO`. As long as the peer count stays high, I don't think the user needs to be actively concerned about how we're responding to peers.

## Additional Info

NA

### CODE REVIEW

File: beacon_node/network/src/beacon_processor/worker/rpc_methods.rs [hunk 1]
```diff
@@ -392,12 +392,26 @@ impl<T: BeaconChainTypes> Worker<T> {
...
-                                "error" => ?e
-                            );
+                            if matches!(
+                                e,
+                                BeaconChainError::ExecutionLayerErrorPayloadReconstruction(_block_hash, ref boxed_error)
+                                if matches!(**boxed_error, execution_layer::Error::EngineError(_))
+                            ) {
+                                warn!(
+                                    self.log,
+                                    "Error rebuildinging payload for peer";
```
Comment (michaelsproul) [inline beacon_node/network/src/beacon_processor/worker/rpc_methods.rs]:
```suggestion
                                    "Error rebuilding payload for peer";
```

**Review by michaelsproul — APPROVED** [review]
LGTM!

File: beacon_node/network/src/beacon_processor/worker/rpc_methods.rs (hunk 1 above)
Comment (paulhauner) [reply beacon_node/network/src/beacon_processor/worker/rpc_methods.rs]: Thanks 😅

**Comment by paulhauner** [comment]
Thanks! ☺️

bors r+

**Comment by paulhauner** [comment]
bors r-

Gonna do some batchin'

**Pull request closed without merging**